docs/upgrade.html

```diff
@@ -45,6 +45,22 @@ <h5><a id="upgrade_240_notable" href="#upgrade_240_notable">Notable changes in 2
         The old overloaded functions are deprecated and we would recommend users to make their code changes to leverage the new methods (details
         can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-520%3A+Add+overloaded+Consumer%23committed+for+batching+partitions">KIP-520</a>).
     </li>
+    <li>We are introducing incremental cooperative rebalancing to the clients' group protocol, which allows consumers to keep all of their assigned partitions during a rebalance
+        and at the end revoke only those which must be migrated to another consumer for overall cluster balance. To upgrade from eager to cooperative rebalancing, you must set all your clients
+        to use the new <code>CooperativeStickyAssignor</code>. This can be done with two rolling bounces: during the first one, add "cooperative-sticky" to the list of supported assignors
+        for each member (without removing the previous assignor -- note that if previously using the default, you must include that explicitly as well) before you bounce and/or upgrade it.
+        Once the entire group is on 2.4+ and all members have the <code>CooperativeStickyAssignor</code> among their supported assignors, perform a second rolling bounce and remove the
+        other assignor(s) so that by the end all members support only the cooperative protocol. For further details on the safe upgrade path or the cooperative rebalancing protocol,
+        see <a href="https://cwiki.apache.org/confluence/x/vAclBg">KIP-429</a>.
+    </li>
+    <li>There are some behavioral changes to the <code>ConsumerRebalanceListener</code>, as well as a new API. The <code>onPartitionsLost</code> method has been added to allow
+        users to react to abnormal circumstances where a consumer may have lost ownership of its partitions (such as a missed rebalance) and cannot commit offsets. By default, this will simply call
+        the existing <code>onPartitionsRevoked</code> API to align with previous behavior. Note however that <code>onPartitionsLost</code> will not be called when the set of lost partitions is empty.
```

Comment:
... however that onPartitionsLost and onPartitionsRevoked ...

EDIT: I saw you mentioned about it below, see my other comment for explicitly explaining the diff of semantics.

Reply:
Yes, I'm just advocating that we should make clear the triggering semantics change with the new mode.